modal-go/test/sandbox_test.go

```diff
@@ -646,3 +647,72 @@ func TestNamedSandboxNotFound(t *testing.T) {
 	g.Expect(err).Should(gomega.HaveOccurred())
 	g.Expect(err.Error()).To(gomega.ContainSubstring("not found"))
 }
+
+func TestNoGoroutineLeaksWithIgnoredOutput(t *testing.T) {
+	testCases := []struct {
+		name      string
+		params    *modal.SandboxExecParams
+		afterExec func(*gomega.WithT, *modal.ContainerProcess)
+	}{
+		{
+			name: "with modal.Ignore",
+			params: &modal.SandboxExecParams{
+				Stdout: modal.Ignore,
+				Stderr: modal.Ignore,
+			},
+			afterExec: func(g *gomega.WithT, p *modal.ContainerProcess) {},
+		},
+		{
+			name:   "with explicit Close",
+			params: &modal.SandboxExecParams{},
+			afterExec: func(g *gomega.WithT, p *modal.ContainerProcess) {
+				err := p.Stdout.Close()
+				g.Expect(err).ShouldNot(gomega.HaveOccurred())
+				err = p.Stderr.Close()
+				g.Expect(err).ShouldNot(gomega.HaveOccurred())
+			},
+		},
+	}
+
+	for _, tt := range testCases {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			g := gomega.NewWithT(t)
+			ctx := context.Background()
+
+			app, err := tc.Apps.FromName(ctx, "libmodal-test", &modal.AppFromNameParams{CreateIfMissing: true})
+			g.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+			image := tc.Images.FromRegistry("alpine:3.21", nil)
+
+			runtime.GC()
+			time.Sleep(100 * time.Millisecond)
+			initialGoroutines := runtime.NumGoroutine()
```

Comment:
Can we grab the init goroutines at the beginning?

```suggestion
			initialGoroutines := runtime.NumGoroutine()

			app, err := tc.Apps.FromName(ctx, "libmodal-test", &modal.AppFromNameParams{CreateIfMissing: true})
			g.Expect(err).ShouldNot(gomega.HaveOccurred())

			image := tc.Images.FromRegistry("alpine:3.21", nil)
```

And then check later with a `Eventually`:


```golang
g.Expect(sandboxExitCode).To(gomega.Equal(0))
g.Eventually(func() int {
    runtime.GC()
	return runtime.NumGoroutine()
}, 5*time.Second, 100*time.Millisecond).Should(gomega.Equal(initialGoroutines))
```

I think removing all the timing + GC code would make the test less flaky.